AmethystAPI/src/minecraft/src/common/world/level/BlockSource.hpp

```diff
@@ -42,84 +60,201 @@ class BlockSource {
     virtual bool containsAnyLiquid(const AABB& box) const;
     virtual bool containsMaterial(const AABB& box, MaterialType material) const;
     virtual bool isUnderWater(const Vec3& pos, const Block& block) const;
+    /**
+     * @brief Gets the Material of a block
+     *
+     * @param pos The position of the block as a BlockPos
+     *
+     * @return Returns a constant Material object
+     */
     virtual const Material& getMaterial(const BlockPos& pos) const;
+    /**
+     * @brief Gets the Material of a block
+     *
+     * @param x The x position of the block as an int
+     * @param y The y position of the block as an int
+     * @param z The z position of the block as an int
+     *
+     * @return Returns a constant Material object
+     */
```

Comment:
I'm not going to deny people adding stuff like this, but is this actually a valuable thing to spend time on? I understand documenting more weird / neiche functions, but these docs don't particularly add anything of value that you couldn't just get by reading the function.

Reply:
I understand that this isn't very valuable.. but this should rather be an exanple on how to document. The dimension class has a good example of how it should actually be used in production. It should add valuable information  like the default sea level.